emcc.py

```diff
@@ -935,8 +940,39 @@ def detect_fixed_language_mode(args):
                 break
             if found: break
           if found: break
-        if not found and lib not in ['GL', 'GLU', 'glut', 'm', 'c', 'SDL', 'stdc++', 'pthread']: # whitelist our default libraries
-          logging.warning('emcc: cannot find library "%s"', lib)
+        if not found:
+          # Some native libraries are implemented in Emscripten as system side JS libraries
+          js_system_libraries = {
```

Comment:
how about refactoring this js system lib logic into `Building` in `tools/shared.py`?